src/firstSuggestionDecoration.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  DecorationOptions,
+  DecorationRangeBehavior,
+  ExtensionContext,
+  InlineCompletionList,
+  Position,
+  Range,
+  TextDocument,
+  TextEditor,
+  window,
+} from "vscode";
+import {
+  firstSuggestionDecorationAlreadyDisplayed,
+  setFirstSuggestionFlag,
+} from "./firstSuggestionDecorationFlag";
+import { fireEvent, getState } from "./binary/requests/requests";
+import isInTheLastHour from "./utils/time.utils";
+import { getTabnineExtensionContext } from "./globals/tabnineExtensionContext";
+import { setDecoration } from "./vscode.api";
+import TabnineInlineCompletionItem from "./inlineSuggestions/tabnineInlineCompletionItem";
+import { MAX_SMALL_INTEGER_V8 } from "./globals/consts";
+import { Capability, isCapabilityEnabled } from "./capabilities/capabilities";
+// Anchor hint to the right of the line
+export const firstSuggestionDecorationType = window.createTextEditorDecorationType(
+  {
+    rangeBehavior: DecorationRangeBehavior.ClosedOpen,
+  }
+);
+
+export function getFirstCompletionDecoration(
+  document: TextDocument,
+  position: Position
+): DecorationOptions[] {
+  const decorations: DecorationOptions[] = [];
+  decorations.push({
+    renderOptions: {
+      after: {
+        color: "gray",
+        contentText: "👈 Press Tab to accept Tabnine's suggestion",
+        margin: "20px",
+        border: "0.5px solid",
+      },
+    },
+    range: document.validateRange(
+      new Range(
+        position.line,
+        MAX_SMALL_INTEGER_V8,
+        position.line,
+        MAX_SMALL_INTEGER_V8
+      )
+    ),
+  });
+  return decorations;
+}
+
+export function displayFirstSuggestionDecoration(position: Position): void {
+  const editor = window.activeTextEditor;
+  if (!editor) {
+    return;
+  }
+  setDecoration(
+    firstSuggestionDecorationType,
+    getFirstCompletionDecoration(editor.document, position),
+    editor
+  );
+  // for now: clear notification after 5 seconds
+  setTimeout(clearFirstSuggestionDecoration, 5000, editor);
```

Comment:
i think you need to clear it on the selection change event as well. What would happen if you press the down-arrow key while the hint is shown?

Reply:
You're right, arrow down leaves the decorator in its place. However, the `window.onDidChangeTextEditorSelection` handler doesn't fire at all when you have an inline suggestion shown and you press the down arrow, since this seems to be handled entirely by the inline completion provider API...